### PR TITLE
ERC-721 support

### PR DESCRIPTION
**Pull request by AlexSaplin**

### CODE REVIEW

**Review by YusupovAI — APPROVED** [review]

**Review by k1rill-fedoseev — CHANGES_REQUESTED** [review]

File: test/Vault.t.sol [hunk 1]
```diff
@@ -117,12 +119,12 @@ contract VaultTest is Test, SetupContract, Utilities {
     // openVault
 
     function testOpenVaultSuccess() public {
-        uint256 oldLen = getLength(vault.ownedVaultsByAddress(address(this)));
```
[the remark this answers is not in the document]
Comment (AlexSaplin) [reply test/Vault.t.sol:121]: https://github.com/zkBob/stable-nft/pull/52/files#diff-52771dcedd958ba5103f0178ee321b06c2781b6f3c3989362302a24358342afbR181

File: src/Vault.sol [hunk 2]
```diff
@@ -222,6 +216,17 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver {
...
 
+    /// @inheritdoc IERC165
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(AccessControlEnumerable, ERC721Enumerable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId) || type(IERC721Receiver).interfaceId == interfaceId;
```
[the remark this answers is not in the document]
Comment (AlexSaplin) [reply src/Vault.sol:232]: fixed

File: src/Vault.sol [hunk 3]
```diff
@@ -746,13 +758,8 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver {
...
     /// @notice Close a vault (internal)
     /// @param vaultId Id of the vault
-    /// @param owner Vault owner
     /// @param nftsRecipient Address to receive nft of the positions in the closed vault
-    function _closeVault(
-        uint256 vaultId,
-        address owner,
-        address nftsRecipient
-    ) internal {
+    function _closeVault(uint256 vaultId, address nftsRecipient) internal {
```
Comment (YusupovAI) [inline src/Vault.sol:762]: I am not sure, why the nft isn't burned here?

Comment (AlexSaplin) [reply src/Vault.sol:762]: We don't have to burn it anywhere in the contract. A user can deposit collateral in the vault after it's closure

**Review by YusupovAI — APPROVED** [review]

**Review by k1rill-fedoseev — APPROVED** [review]

**Pull request merged**